priv/stdlib/List.aes

```diff
@@ -37,14 +37,18 @@ namespace List =
     []   => reverse(acc)
     h::t => find_indices_(p, t, n+1, if(p(h)) n::acc else acc)
 
-  function nth(n : int, l : list('a)) : option('a) = switch(l)
-    []   => None
-    h::t => if(n == 0) Some(h) else nth(n-1, t)
+  function nth(n : int, l : list('a)) : option('a) =
+    if(n < 0) None
+    else switch(l)
+      []   => None
+      h::t => if(n == 0) Some(h) else nth(n-1, t)
 
   /* Unsafe version of `nth` */
-  function get(n : int, l : list('a)) : 'a = switch(l)
-    [] => abort("Out of index get")
-    h::t => if(n == 0) h else get(n-1, t)
+  function get(n : int, l : list('a)) : 'a =
+    if(n < 0) abort("Negative index get")
+    else switch(l)
+      [] => abort("Out of index get")
+      h::t => if(n == 0) h else get(n-1, t)
```

Comment:
Same here. It doesn't seem worth the extra check just to get a special error for negative indices.

Reply:
Moved `if` to the `abort`. It will keep meaningful error message and not affect performance